src/client/Client.h

```diff
@@ -1299,4 +1335,98 @@ class StandaloneClient : public Client
   void shutdown() override;
 };
 
+class destructive_lock_ref_t {
+public:
+  destructive_lock_ref_t(const destructive_lock_ref_t &) = delete;
+  destructive_lock_ref_t(const destructive_lock_ref_t &&) = delete;
+  destructive_lock_ref_t(class Client *c, Client::_state state, bool update_state) {
```

Comment:
Why the bool argument here? It's not used.

Reply:
Locally I have update it and will fold the two constructors into one with a default value.